ush/oceanice_nc2grib2.sh

```diff
@@ -0,0 +1,319 @@
+
+}
+
+################################################################################
+function _ocean3D_nc2grib2 {
+# This function converts the ocean 3D rectilinear netCDF files to grib2 format
+
+  # Set the inputs
+  local grid=${1} # 0p25, 0p50, 1p00, 5p00
+  local latlon_dims=${2} # 0:721:0:1440, 0:361:0:720, 0:181:0:360, 0:36:0:72
+  local levels=${3} # 5:15:25:35:45:55:65:75:85:95:105:115:125
+  local vdate=${4} # YYYYMMDDHH
+  local aperiod=${5} # 0-6
+  local infile=${6} # ocean.0p25.nc
+  local outfile=${7} # ocean_3D.0p25.grib2
+  local template=${8} # template.global.0p25.gb2
+
+  IFS=':' read -ra depths <<< "${levels}"
+
+  zl=0
+  for depth in "${depths[@]}"; do
+
+    [[ -f "tmp.gb2" ]] && rm -f "tmp.gb2"
+
+    ${WGRIB2} "${template}" \
+    -import_netcdf "${infile}" "temp" "0:1:${zl}:1:${latlon_dims}" \
+      -set_var WTMP -set center 7 -rpn "273.15:+" \
+      -set_lev "${depth} m below water surface" \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out tmp.gb2 \
+    -import_netcdf "${infile}" "so" "0:1:${zl}:1:${latlon_dims}" \
+      -set_var SALTY -set center 7 -rpn "1000.0:/" \
+      -set_lev "${depth} m below water surface" \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out tmp.gb2 \
+    -import_netcdf "${infile}" "uo" "0:1:${zl}:1:${latlon_dims}" \
+      -set_var UOGRD -set center 7 \
+      -set_lev "${depth} m below water surface" \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out tmp.gb2 \
+    -import_netcdf "${infile}" "vo" "0:1:${zl}:1:${latlon_dims}" \
+      -set_var VOGRD -set center 7 \
+      -set_lev "${depth} m below water surface" \
+      -set_date "${vdate}" -set_ftime "${aperiod} hour ave fcst" \
+      -set_scaling same same -set_grib_type c1 -grib_out tmp.gb2
+
+    rc=$?
+    # Check if the conversion was successful
+    if (( rc != 0 )); then
+      echo "FATAL ERROR: Failed to convert the ocean rectilinear netCDF file to grib2 format at depth ${depth}m, ABORT!"
+      return "${rc}"
+    fi
+
+    cat tmp.gb2 >> "${outfile}"
+    rm -f tmp.gb2
+    ((zl = zl + 1))
+
+  done
+
+  # Notes:
+  #   WATPTEMP (water potential temperature (theta)) may be a better
+  #   GRIB2 parameter than WTMP (water temperature) if MOM6 outputs
+  #   potential temperature. WATPTEMP is not available in NCEP
+  #   (-set center 7) tables in wgrib2 v2.0.8.
+
+  return "${rc}"
+
+}
+
+################################################################################
+# Input arguments
+component=${1:?"Need a valid component; options: ice|ocean"}
+grid=${2:-"0p25"} # Default to 0.25-degree grid
+vdate=${3:-"2013100100"} # Default to 2013100100
+avg_period=${4:-"0-6"} # Default to 6-hourly average
+ocean_levels=${5:-"5:15:25:35:45:55:65:75:85:95:105:115:125"} # Default to 12-levels
```

Comment:
Those are just for default in case a 5th argument is not passed.
The ocean levels are defined in the [oceanice_products.yaml](https://github.com/NOAA-EMC/global-workflow/pull/2286/files#diff-3ec43883531d5cd1250fd1dea76d6ffc99bdb90dafbee6b20b6dbcfbafcf07b5).
The output log from the test experiment shows all 40 depths were processed.
You can look at the run log on Hera at:
`/scratch1/NCEPDEV/stmp2/Rahul.Mahajan/RUNTESTS/COMROOT/oi2/logs/2016070100/gfsoceanprod_f114-f120.log`
Attached is a screenshot for the log showing that the loop did reach the last depth value at `4481.0625m`.
![Screenshot 2024-02-05 at 3 41 00 PM](https://github.com/NOAA-EMC/global-workflow/assets/11394126/0034a7c9-02b2-40cb-b3e5-27039d5e1133)